base/asyncmap.jl

```diff
@@ -22,17 +23,17 @@ end
 
 function AsyncCollector(f, results, c...; ntasks=0)
     if ntasks == 0
-        ntasks = 100
+        ntasks = max(nworkers(), 100)
```

Comment:
The above fix is still required.

As for your error, googling brought up http://www.toptip.ca/2010/02/linux-eaddrnotavail-address-not.html which should not happen with only a 100 workers. But you could test the range of temporary ports available as explained on the page to see if it is abysmally low.